src/api/File.js

```diff
@@ -78,29 +79,71 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * Polls an item's infoUrl until it's representation is generated or an error occurrs.
+     * Calls callback if representation was successfully generated.
+     *
+     * @param {BoxItem} item - BoxItem that needs infoUrl polled
+     * @param {string} thumbanilUrl - thumbanil url of item
+     * @param {string, string) => void} callback - function to call if polling results in success
+     * @return {Promise<void>}
+     */
+    async pollInfoUrl(
+        item: BoxItem,
+        thumbnailUrl: string,
+        callback: (id: string, thumbnailUrl: string) => void = noop,
+    ): Promise<void> {
+        const { id } = item;
+        const infoUrl = getProp(item, 'representations.entries[0].info.url');
+
+        if (!infoUrl) {
+            return;
+        }
+
+        let status;
+        do {
+            // eslint-disable-next-line no-await-in-loop
+            status = getProp(await this.xhr.get({ url: infoUrl }), 'data.status.state');
```

Comment:
Would it be worth it to put in some kind of delay here rather than constantly polling the url to see if the thumbnail is generated?

Reply:
As discussed, check out `retryNumOfTimes` from `utils/function`

Reply:
Looking into using `retryNumOfTimes` in `utils/function` to do an exponential backoff on requests.